src/cdk-experimental/menu/menu-item-trigger.ts

```diff
@@ -24,29 +42,122 @@ import {CdkMenuItem} from './menu-item';
   exportAs: 'cdkMenuTriggerFor',
   host: {
     'aria-haspopup': 'menu',
+    '[attr.aria-expanded]': 'isSubmenuOpen()',
   },
 })
-export class CdkMenuItemTrigger implements AfterContentInit {
+export class CdkMenuItemTrigger implements OnDestroy {
   /** Template reference variable to the menu this trigger opens */
   @Input('cdkMenuTriggerFor') _menuPanel?: CdkMenuPanel;
 
+  /** Emits when the attached submenu is requested to open */
+  @Output() readonly cdkMenuOpened: EventEmitter<void> = new EventEmitter();
+
+  /** Emits when the attached submenu is requested to close  */
+  @Output() readonly cdkMenuClosed: EventEmitter<void> = new EventEmitter();
+
+  /** A reference to the overlay which manages the triggered submenu */
+  private _overlayRef: OverlayRef | null = null;
+
+  /** The Portal in which the Menu is displayed inside of */
+  private _portal: TemplatePortal;
+
   constructor(
-    /** The MenuItem instance which is the trigger  */
-    @Self() @Optional() private _menuItemInstance?: CdkMenuItem
+    private readonly _elementRef: ElementRef<HTMLElement>,
+    protected readonly _viewContainerRef: ViewContainerRef,
+    private readonly _overlay: Overlay,
+    private readonly _directionality: Directionality,
+    @Inject(CDK_MENU) private readonly _parentMenu: Menu
   ) {}
 
-  ngAfterContentInit() {
-    this._setHasSubmenu();
-  }
-
-  /** Set the hasSubmenu property on the menuitem  */
-  private _setHasSubmenu() {
-    if (this._menuItemInstance) {
-      this._menuItemInstance.hasSubmenu = this._hasSubmenu();
+  /** Open/close the attached submenu if the trigger has been configured with one */
+  toggle() {
+    if (this.hasSubmenu()) {
+      this.isSubmenuOpen() ? this._closeSubmenu() : this._openSubmenu();
     }
   }
+
   /** Return true if the trigger has an attached menu */
-  private _hasSubmenu() {
+  hasSubmenu() {
```

Comment:
Thinking about this more, should this just be `hasMenu` and `isMenuOpen` below? Each trigger is inherently connected to a single menu item, regardless of whether it's a top-level menu or a sub-menu on a menu item (I think the class name also then becomes just `CdkMenuTrigger`)

Reply:
Since `cdkMenuTrigger` is added alongside `cdkMenuItem` would CdkMenuItem also drop the `Sub`?

I think it makes sense to stick with one approach for consistency since it becomes confusing why a MenuItem has a submenu and a trigger has a Menu.

Reply:
SGTM to drop "sub" everywhere in a follow-up PR